flutter/news/test/features/frontpage/data/repositories/articles_repository_test.dart

```diff
@@ -24,128 +20,78 @@ import 'articles_repository_test.mocks.dart';
-          title: "title",
-          description: "description",
-          url: "url",
-          urlToImage: "urlToImage",
-          publishedAt: "publishedAt",
-          content: "content")
-    ];
-    final List<Article> tArticleList = tArticlesModelList;
-
-    runTestsOnline((){
-
-      test(
-        'should return remote data when call to remote data is success',
-        () async {
-          when(mockArticlesRemoteDataSource.getTopHeadlines())
-              .thenAnswer((realInvocation) async => tArticlesModelList);
-
-          final result = await repository.getTopHeadlines();
-
-          verify(mockArticlesRemoteDataSource.getTopHeadlines());
-          expect(result, Right(tArticleList));
-        },
-      );
-
-      test(
-        'should return server failure when call to remote data fails',
-        () async {
-          when(mockArticlesRemoteDataSource.getTopHeadlines())
-              .thenThrow(ServerException("exception"));
-
-          final result = await repository.getTopHeadlines();
-
-          verify(mockArticlesRemoteDataSource.getTopHeadlines());
-          verifyZeroInteractions(mockArticlesLocalDataSource);
-          expect(result, const Left(ServerFailure("exception")));
-        },
-      );
-
-      test(
-        'should cache data locally when call to remote data is success',
-        () async {
-          when(mockArticlesRemoteDataSource.getTopHeadlines())
-              .thenAnswer((realInvocation) async => tArticlesModelList);
-
-          await repository.getTopHeadlines();
-
-          verify(mockArticlesRemoteDataSource.getTopHeadlines());
-          verify(mockArticlesLocalDataSource
-              .cacheTopHeadlines(tArticlesModelList));
-        },
-      );
-    });
-
-    runTestsOffline((){
-      test(
-        'should return cached data when device is offline && has cache data',
-        () async {
-          when(mockArticlesLocalDataSource.getLastTopHeadlines())
-              .thenAnswer((realInvocation) async => tArticlesModelList);
-
-          final result = await repository.getTopHeadlines();
-
-          verifyNoMoreInteractions(mockArticlesRemoteDataSource);
-          verify(mockArticlesLocalDataSource.getLastTopHeadlines());
-          expect(result, Right(tArticleList));
-        },
-      );
-
-      test(
-        'should return cached failure when there is no cached values',
-        () async {
-          when(mockArticlesLocalDataSource.getLastTopHeadlines())
-              .thenThrow(CacheException("cacheException"));
-
-          final result = await repository.getTopHeadlines();
-
-          verifyNoMoreInteractions(mockArticlesRemoteDataSource);
-          verify(mockArticlesLocalDataSource.getLastTopHeadlines());
-          expect(result, const Left(CacheFailure("cacheException")));
-        },
-      );
-    });
-  });
+  test(
+    'GIVEN remote fetch will succeed WHEN getting top articles THEN saves articles on local AND returns local articles',
```

Comment:
@Diogo00dev what do you think about this way of writing tests?

Reply:
agree, i think its more intuitive